Fix build errors on AVX2+ hosts with -march=native.

pwnall opened the pull request:
third_party/eigen3/unsupported/Eigen/CXX/FixedPoint and third_party/eigen3/unsupported/Eigen/CXX/Tensor use different paths to reach into unsupported/Eigen/CXX/src/Tensor. The former's paths cause the build errors documented in #6558. This commit converges on the path in third_party/eigen3/unsupported/Eigen/CXX/Tensor, which fixes the build error.

I tested the PR with the following command, which fails without the patch.

```bash
bazel build -c opt --copt=-march=native //tensorflow:libtensorflow.so
```

pwnall committed:
Fix build errors on AVX2+ hosts with -march=native.

third_party/eigen3/unsupported/Eigen/CXX/FixedPoint and
third_party/eigen3/unsupported/Eigen/CXX/Tensor use different paths to
reach into unsupported/Eigen/CXX/src/Tensor. The former's paths cause
the build errors documented in GitHub issue #6558. This commit converges
on the path in third_party/eigen3/unsupported/Eigen/CXX/Tensor, which
fixes the build error.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

drpngx commented:
Could you add a comment immediately preceding the change explaining the problem succintly?

pwnall commented:
@drpngx I hope I have addressed your feedback. PTAL?

I also submitted #6759, which may be a better fix. I don't have access to all the tensorflow buildbots though, so I'm not sure it doesn't break anything. Either that PR or this one should take care of #6558.

Thank you very much for your review!

drpngx commented:
Jenkins, test this please.

Please ignore the estimator_test error.

drpngx commented:
If the ThreadPool include is genuinely used in that file, then this is a better fix. 

It sounds like we should try to push upstream fixes to eigen as @fnl suggested, then undo this change once propagated.

pwnall commented:
@drpngx Are you referring to [this comment](https://github.com/tensorflow/tensorflow/issues/6558#issuecomment-269643996) or another comment from @fnl?

FWIW, I was able to get a full build using `bazel build -c opt --copt=-march=native //...` on an AVX2 machine, so I'm somewhat skeptical that the Tensor include is needed in the FixedNum file. I'm working on getting a Skylake build going (I'm on OSX, and Apple's clang crashes on `--copt=-march=skylake`).

drpngx commented:
Yes, it sounds like it's something with Eigen that should be fixed there, ultimately.

pwnall commented:
My understanding is that @fnl is saying that if TensorContractionThreadPool.h is copied from eigen3 into third_party/eigen3/ then the build succeeds. To me, this seems an indication that the path to reach it is incorrect.

Or, are you saying that we shouldn't have to ship a modified FixedPoint header at all?

drpngx commented:
OK, it looks like #6759 is the better fix. Could you put a comment in that file to explain that we can't add the include in this place?

kunthar commented:
@drpngx since you've triggered one of our remote sensors with the word "Jenkins" + "please", your message arrived to our space time. Please stand by in your coordinates, we will beam you up to our star system soon. Thank you.
